### PR TITLE
Allow use cpu_serial_kernel with void-lambda

### PR DESCRIPTION
**Pull request by ifedan**
Currently we use CPU_tensor_apply1 to loop through the tensor in single thread and aggregate data:
```
// compute variance per input
 accscalar_t var_sum = 0;
 CPU_tensor_apply1<scalar_t>(in, [&] (const scalar_t& i) {
    var_sum += (i - mean) * (i - mean);
 });
```
and we don't have the ability to use TensorIterator for this.

```
accscalar_t var_sum = 0;
auto iter = TensorIterator::unary_op(self, self);
  cpu_serial_kernel(iter, [&](scalar_t i) -> scalar_t {
        var_sum += (i - mean) * (i - mean);
  return a; //Unable to set value back, because self should be const
});
```

This PR should resolve this problem and allow to use void-lambda:
```
auto iter = at::TensorIterator();
iter.add_input(in);
iter.build();
accscalar_t var_sum = 0;                                                            \
at::native::cpu_serial_kernel(iter, [&](scalar_t i) -> void { 
   var_sum += (i - mean) * (i - mean);
});
```

In the future it make sense to change Reduction part and allow to reduce to a scalar, not just to a tensor

### CODE REVIEW

File: aten/src/ATen/native/cpu/Loops.h [hunk 1]
```diff
@@ -217,6 +239,7 @@ void cpu_serial_kernel(TensorIterator& iter, func_t op) {
       });
     }
   }, {0, iter.numel()});
+  iter.cast_outputs();
```
Comment (VitalyFedyunin) [inline aten/src/ATen/native/cpu/Loops.h]: Unrelated change, right?

**Review by VitalyFedyunin — APPROVED** [review]
Please benchmark any existing cpu_serial_kernel op before landing.

File: aten/src/ATen/native/cpu/Loops.h (hunk 1 above)
Comment (ifedan) [reply aten/src/ATen/native/cpu/Loops.h]: yes, it was not added when introduced type promotion.

Comment (VitalyFedyunin) [reply aten/src/ATen/native/cpu/Loops.h]: Good to go with it. Just making sure.

**Comment by ezyang** [comment]
This broke PyTorch windows
